src/core/net/mdns.cpp

```diff
@@ -2948,6 +2985,18 @@ void Core::TxMessage::AppendServicesDnssdName(Section aSection)
     return;
 }
 
+void Core::TxMessage::AddQuestionFrom(OwnedPtr<Message> &aMessage)
```

Comment:
Can change the input to be `const Message &` (since we are only reading from it)?

`OwnePtr<>` is often used when we are allowing ownership trasnfers.